### DISCUSSION TITLE
Deep Cloning

### DISCUSSION THREAD

**Discussion by fredericoregateiro** [riok/mapperly#420] [Q&A] [answered]
Hi everyone,

Could mapperly be used to deep clone one object?
Something like this:

``` cs
[Mapper(UseDeepCloning = true)]
public partial class DocumentMapper
{
    public partial Document Map(Document doc);
}

public class Document
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
}
```

The code generated is like this:

``` cs
public partial global::ConsoleAppMapperly.Document? Map(global::ConsoleAppMapperly.Document? doc)
{
    if (doc == null)
        return default;
    var target = new global::ConsoleAppMapperly.Document();
    target.Name = doc.Name;
    target.Description = doc.Description;
    return target;
}
```

The `Id` is not mapped, why?

**Comment by TimothyMakkison** [comment]
That's strange I can't replicate this. Which version of mapperly are you using? 
```c#
// Assuming nullable is disabled
public partial class DocumentMapper
{
    public partial global::Riok.Mapperly.Sample.Document? Map(global::Riok.Mapperly.Sample.Document? doc)
    {
        if (doc == null)
            return default;
        var target = new global::Riok.Mapperly.Sample.Document();
        target.Id = doc.Id;
        target.Name = doc.Name;
        target.Description = doc.Description;
        return target;
    }
}
```

**Reply by fredericoregateiro** [reply]
mapperly version 2.8.0 in a .net framework 4.8 project

**Reply by fredericoregateiro** [reply]
Just tested width different targets and it works with .net 7.0 and netstandard2.1 but it doesn't work on netstandard2.0 and .net 4.8

**Reply by TimothyMakkison** [reply]
Thanks for the update. I've managed to replicated your bug 👍 I'll try some different versions of Mapperly and then try to find the issue.

Update: tested 2.8, 2.7 and 2.6 and they all fail. I'm suprised an integration test didn't catch this.

**Reply by TimothyMakkison** [reply]
Thanks again for the help. This issue is known and has been fixed via #388, see also #362. 

You can find the preview version with the fix [here](https://www.nuget.org/packages/Riok.Mapperly/2.8.0-next.2).

**Reply by fredericoregateiro** [reply]
I got the same result with version 2.8.0-next.2.
As a side note, the nuget manager shows 2.8.0-next.2 as a previous version of 2.8.0

**Reply by latonz** [reply, marked as answer]
The fix for this bug is merged into main but not yet released. A preview release will be published in the upcoming days including the fix in question.

**Reply by TimothyMakkison** [reply]
Ah yeah, mb. I think VS caching tricked me 😐

**Reply by latonz** [reply]
`2.9.0-next.1` is now released including this bug fix.

**Reply by fredericoregateiro** [reply]
Just tested the 2.9.0-next.1, and it wasn't working until i restarted VS 🔮 

Now it's working in .net framework 4.8, .net 7 and .net standard 2.0 and producing the same code 👏
